runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/SplittableDoFnOperator.java

```diff
@@ -126,7 +127,12 @@ public void initializeState(StateInitializationContext context) throws Exception
     // this will implicitly be keyed like the StateInternalsFactory
     TimerInternalsFactory<byte[]> timerInternalsFactory = key -> timerInternals;
 
-    executorService = Executors.newSingleThreadScheduledExecutor(Executors.defaultThreadFactory());
+    if (this.executorService == null) {
+      this.executorService =
+          Executors.newScheduledThreadPool(
+              Runtime.getRuntime().availableProcessors(),
```

Comment:
I think these other ones could stay a single thread, since it was already shared.

Reply:
changed back to newSingleThreadScheduledExecutor